### PR TITLE
feat: consumer access limit to some routes for scopes

### PR DESCRIPTION
**Pull request by richardlt**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12336.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests -   
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12345.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12347.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12350.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12354.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12357.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Helm Tests -   
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12360.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: engine/api/router.go [hunk 1]
```diff
@@ -182,6 +184,74 @@ func DefaultHeaders() map[string]string {
...
+			for j := range methods {
+				m[scopes[i]][uri][methods[j]] = struct{}{}
+			}
+		}
+	}
+
+	// return scope details
+	details := make([]sdk.AuthConsumerScopeDetail, len(sdk.AuthConsumerScopes))
+	for i, scope := range sdk.AuthConsumerScopes {
+		var endpoints []sdk.AuthConsumerScopeEndpoint
```
Comment (bnjjj) [inline engine/api/router.go]: missing make

File: ui/src/app/views/settings/user/consumer-create-modal/consumer-create-modal.html [hunk 2]
```diff
@@ -4,49 +4,109 @@
...
+                                </sui-accordion-panel>
+                                <sui-accordion-panel [isOpen]="activeStep === formStepName.GROUPS" [isDisabled]="true">
+                                    <div title (click)="clickOpenStep(formStepName.GROUPS)">
+                                        <i class="dropdown icon"></i>
+                                        {{'user_auth_groups' | translate}}
+                                    </div>
+                                    <div content>
+                                        <div class="ui form" *ngIf="!newConsumer.id">
+                                            <div class="ui message blue">
+                                                <p>Let groups selection empty to create consumer with wildcard access on
```
Comment (bnjjj) [inline ui/src/app/views/settings/user/consumer-create-modal/consumer-create-modal.html]: Translation

File: ui/src/app/views/settings/user/consumer-create-modal/scope-detail.component.ts [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        this.allWriteRoutesSelected = allWriteRoutesSelected;
+    }
+
+    syncGlobalAllCheckbox() {
+        // check if all routes are selected
+        let allRoutesSelected = true;
+        const routes = Object.keys(this.selectedRouteMethods);
+        for (let i = 0; i < routes.length; i++) {
+            if (!this.selectedRouteMethods[routes[i]].ALL) {
+                allRoutesSelected = false;
```
Comment (bnjjj) [inline ui/src/app/views/settings/user/consumer-create-modal/scope-detail.component.ts]: missing break ?

File: ui/src/app/views/settings/user/consumer-create-modal/scope-detail.component.ts [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        this.allReadRoutesSelected = allReadRoutesSelected;
+    }
+
+    syncGlobalWriteCheckbox() {
+        // check if all write routes are selected
+        let allWriteRoutesSelected = true;
+        const routes = Object.keys(this.selectedRouteMethods);
+        for (let i = 0; i < routes.length; i++) {
+            if (this.selectedRouteMethods[routes[i]].POST != null && !this.selectedRouteMethods[routes[i]].POST) {
+                allWriteRoutesSelected = false;
```
Comment (bnjjj) [inline ui/src/app/views/settings/user/consumer-create-modal/scope-detail.component.ts]: break ?

File: ui/src/app/views/settings/user/consumer-create-modal/scope-detail.component.ts [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+            }
+            if (this.selectedRouteMethods[routes[i]].PUT != null && !this.selectedRouteMethods[routes[i]].PUT) {
+                allWriteRoutesSelected = false;
```
Comment (bnjjj) [inline ui/src/app/views/settings/user/consumer-create-modal/scope-detail.component.ts]: break ?

File: ui/src/app/views/settings/user/consumer-create-modal/scope-detail.component.ts [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+            }
+            if (this.selectedRouteMethods[routes[i]].DELETE != null && !this.selectedRouteMethods[routes[i]].DELETE) {
+                allWriteRoutesSelected = false;
```
Comment (bnjjj) [inline ui/src/app/views/settings/user/consumer-create-modal/scope-detail.component.ts]: break ?

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12407.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Pull request merged**